dipy/denoise/fast_noise_estimate.pyx

```diff
@@ -0,0 +1,134 @@
+        methods
+
+    Returns
+    -------
+    sigma_corr: 3D array
+        The local noise variance estimate
+
+    References
+    ----------
+    [1]: Manjon JV, Coupe P, Concha L, Buades A, Collins DL
+        "Diffusion Weighted Image Denoising Using Overcomplete
+         Local PCA"
+         
+
+    """
+    
+    # first identify the number of the b0 images
+    K = gtab.b0s_mask[gtab.b0s_mask].size
+
+    if(K > 1):
+        # If multiple b0 values then
+        data0 = data[..., gtab.b0s_mask]
+
+    else:
+        # if only one b0 value then
+        # SIBE Noise Estimate
+        data0 = data[..., ~gtab.b0s_mask]
+        # MUBE Noise Estimate
+
+    cdef:
+        cnp.npy_intp n0 = data0.shape[0]
+        cnp.npy_intp n1 = data0.shape[1]
+        cnp.npy_intp n2 = data0.shape[2]
+        cnp.npy_intp n3 = data0.shape[3]
+        cnp.npy_intp nsamples = n0 * n1 * n2
+        cnp.npy_intp i, j, k, i0, j0, k0, l0
+        double sum_reg, temp1
+        double[:, :, :] I = np.zeros((n0, n1, n2))
+        double[:, :, :] count = np.zeros((n0, n1, n2))
+        double[:, :, :] mean = np.zeros((n0, n1, n2))
+        double[:, :, :] sigma = np.zeros((n0, n1, n2))
+        double[:, :, :, :] data0temp = data0
+
+    X = data0.reshape(
+        data0.shape[0] *
+        data0.shape[1] *
+        data0.shape[2],
+        data0.shape[3])
+    # X = NxK
+    # Now to subtract the mean
+    M = np.mean(X, axis=0)
+    X = X - np.array([M, ] * X.shape[0],
+                                      dtype=np.float64)
+    C = np.transpose(X).dot(X)
+    C = C / X.shape[0]
+    # Do PCA and find the lowest principal component
+    [d, W] = np.linalg.eigh(C)
+    d[d < 0] = 0
+    d_new = d[d != 0]
+    # we want eigen vectors of XX^T so we do
+    V = X.dot(W)
+
+    I = V[:, d.shape[0] - d_new.shape[0]].reshape(
+        data0.shape[0],
+        data0.shape[1],
+        data0.shape[2])
+                
+    with nogil:
+        for i in range(1, n0 - 1):
+            for j in range(1, n1 - 1):
+                for k in range(1, n2 - 1):
+                    sum_reg = 0
+                    temp1 = 0
+                    for i0 in range(-1,2):
+                        for j0 in range(-1,2):
+                            for k0 in range(-1,2):
+                                sum_reg += I[i + i0, j + j0, k + k0] / 27.0
+                                for l0 in range(n3):
+                                    temp1 += (data0temp[i + i0, j+ j0, k + k0, l0]) / (27.0 * n3)
+
+                    for i0 in range(-1,2):
+                        for j0 in range(-1,2):
+                            for k0 in range(-1,2):
+                                sigma[i + i0, j +j0, k + k0] += (I[i + i0, j + j0, k + k0] - sum_reg)**2
+                                mean[i + i0, j + j0, k + k0] += temp1
+                                count[i + i0, j +j0, k + k0] += 1             
+
+    sigma = np.divide(sigma, count)
```

Comment:
The (unbiased for samples) standard deviation needs a N-1 denominator, you seem to use sum(X - mean)**2 / N instead.